src/Raven.Client/Documents/Session/InMemoryDocumentSessionOperations.cs

```diff
@@ -820,6 +830,22 @@ private void PrepareCompareExchangeEntities(SaveChangesData result)
             clusterTransactionOperations.Clear();
         }
 
+        private void PrepareForCountersIncrement(SaveChangesData result)
+        {
+            foreach (var (id, type, name) in result.DeferredCommandsDictionary.Keys)
```

Comment:
Just iterate over the dictionary directly, not on the `Keys`, you can also then directly check the type of the value and not check the type.

Reply:
removed this method